### PR TITLE
 Apply (Login Redirection to Programs) PRs

### PR DESCRIPTION
**Pull request by shadinaif**
This applies the following PRs and adding few tests:

https://github.com/Edraak/edx-platform/pull/300 Enabling login/registration views to redirecet back to other platforms
https://github.com/Edraak/edx-platform/pull/308 Login requests will redirect to the programs platform dashboard

### CODE REVIEW

File: lms/urls.py [hunk 1]
```diff
@@ -55,6 +55,16 @@
 from track import views as track_views
 from util import views as util_views
 
+from course_modes.edraak_helpers import get_progs_url
+
+# Edraak: Redirect dashboard to Programs
+DASHBOARD_VIEW = student_views.student_dashboard
+if settings.PROGS_URLS and settings.PROGS_URLS.get('DASHBOARD'):
+    DASHBOARD_VIEW = RedirectView.as_view(
+        url=get_progs_url(settings.PROGS_URLS.get("DASHBOARD")),
```
Comment (ghost) [inline lms/urls.py]:
```suggestion
        url=get_progs_url(settings.PROGS_URLS['DASHBOARD']),
```

File: lms/envs/aws.py [hunk 2]
```diff
@@ -676,6 +676,12 @@
 
 PROGS_URLS = ENV_TOKENS.get('PROGS_URLS', PROGS_URLS)
 
+# Redirection options
+if FEATURES.get("EDRAAK_ENABLE_AUTH_EXTERNAL_REDIRECT"):
+    EDRAAK_AUTH_REDIRECT_ALLOW_ANY = ENV_TOKENS.get("EDRAAK_AUTH_REDIRECT_ALLOW_ANY", False)
+    EDRAAK_AUTH_REDIRECT_ORIGINS_WHITELIST = ENV_TOKENS.get("EDRAAK_AUTH_REDIRECT_ORIGINS_WHITELIST", tuple())
+    EDRAAK_AUTH_REDIRECT_REGX_ORIGINS = ENV_TOKENS.get("EDRAAK_AUTH_REDIRECT_REGX_ORIGINS", tuple())
```
Comment (ghost) [inline lms/envs/aws.py]:
```suggestion
EDRAAK_AUTH_REDIRECT_REGX_ORIGINS = ENV_TOKENS.get("EDRAAK_AUTH_REDIRECT_REGX_ORIGINS", tuple())
```

Comment (ghost) [reply lms/envs/aws.py]: Please de-indent, same goes for the features above.

File: lms/envs/aws.py [hunk 3]
```diff
@@ -676,6 +676,12 @@
 
 PROGS_URLS = ENV_TOKENS.get('PROGS_URLS', PROGS_URLS)
 
+# Redirection options
+if FEATURES.get("EDRAAK_ENABLE_AUTH_EXTERNAL_REDIRECT"):
```
Comment (ghost) [inline lms/envs/aws.py]:
```suggestion
```

That's an anti-pattern. Please avoid using it. I don't see any benefit of having settings variables appearing conditionally.

File: lms/djangoapps/student_account/views.py [hunk 4]
```diff
@@ -175,6 +176,9 @@ def login_and_registration_form(request, initial_mode="login"):
     enterprise_customer = enterprise_customer_for_request(request)
     update_logistration_context_for_enterprise(request, context, enterprise_customer)
 
+    # Edraak: add (origin) option to the context if applicable
+    edraak_update_origin(request, context)
```
Comment (ghost) [inline lms/djangoapps/student_account/views.py]:
```suggestion
    edraak_update_origin(request, context)  # Warning: This modifies `context['data']['origin']`
```

**Review by ghost — APPROVED** [review]
Few updates and it's good to go. Thanks @shadinaif!

File: lms/urls.py (hunk 1 above)
Comment (shadinaif) [reply lms/urls.py]: Done

File: lms/envs/aws.py (hunk 2 above)
Comment (shadinaif) [reply lms/envs/aws.py]: Done

File: lms/envs/aws.py (hunk 3 above)
Comment (shadinaif) [reply lms/envs/aws.py]: Done

File: lms/djangoapps/student_account/views.py (hunk 4 above)
Comment (shadinaif) [reply lms/djangoapps/student_account/views.py]: Done

**Pull request merged**